packages/react-native/processSourceMap.js

```diff
@@ -0,0 +1,62 @@
+const bundleToString = require('metro/src/lib/bundleToString');
+const baseJSBundle = require('metro/src/DeltaBundler/Serializers/baseJSBundle');
+const CountingSet = require('metro/src/lib/CountingSet').default;
+const fs = require('fs');
+const path = require('path');
+const { DebugIdGenerator, SourceProcessor } = require('@backtrace/sourcemap-tools');
+
+const DEBUG_ID_PATH = process.env.DEBUG_ID_PATH;
+
+/**
+ * Process metro build with source map support powered by Backtrace.
+ */
+function processSourceMap(entryPoint, preModules, graph, options) {
+    const bundle = bundleToString(baseJSBundle(entryPoint, preModules, graph, options));
+
+    // development build - skip source map upload
+    if (graph.transformOptions.hot || graph.transformOptions.dev) {
+        return bundle;
+    }
+
+    // no source map option set
+    const sourceMapOutputPathParameter = process.argv.indexOf('--sourcemap-output');
+    if (sourceMapOutputPathParameter === -1) {
+        return bundle;
+    }
+
+    const sourceMapOutputPath = process.argv[sourceMapOutputPathParameter + 1];
+    const backtraceSourceMapId =
+        DEBUG_ID_PATH ?? path.join(path.dirname(sourceMapOutputPath), '.backtrace-sourcemap-id');
```

Comment:
would it be possible to get `sourceMapOutputPath` in Gradle task as well?

I'm thinking that instead of creating a lone `.backtrace-sourcemap-id` file, you could save it to `${sourceMapOutputPath}.bt-debug-id` or something like that. That way, it'll be more connected to the file.

Reply:
sourceMapOutputPath might change depending on what is being invoked - sourceMapOutputPath is different in the metro parameter, and might be different later in gradle where the user override might be respected.

Reply:
:+1: